miasm2/ir/ssa.py

```diff
@@ -0,0 +1,279 @@
+        if v not in self._stack_lhs:
+            self._stack_lhs[v] = 0
+        # save last value for RHS transformation
+        self._stack_rhs[v] = self._stack_lhs[v]
+
+        # generate SSA expression
+        stack = self._stack_lhs
+        e = self._gen_var_expr(v, stack)
+
+        return e
+
+    def _transform_expression_lhs(self, dst):
+        """
+        Transforms an expression on the left hand side into SSA
+        :param dst: expression
+        :return: expression in SSA form
+        """
+        if isinstance(dst, m2_expr.ExprMem):
+            # transform with last RHS instance
+            e = self._transform_expression_rhs(dst)
+        else:
+            # transform LHS
+            e = self._transform_var_lhs(dst)
+
+            # increase SSA variable counter
+            self._stack_lhs[dst] += 1
+
+        return e
+
+    def _transform_expression_rhs(self, src):
+        """
+        Transforms an expression on the right hand side into SSA
+        :param src: expression
+        :return: expression in SSA form
+        """
+        # dissect expression in variables
+        variables = self._expr_dissect.variables(src)
+        src_ssa = src
+        # transform variables
+        for v in variables:
+            v_ssa = self._transform_var_rhs(v)
+            src_ssa = src_ssa.replace_expr({v: v_ssa})
+
+        return src_ssa
+
+    @staticmethod
+    def _parallel_instructions(assignblk):
+        """
+        Extracts the instruction from a AssignBlock.
+
+        Since instructions in a AssignBlock are evaluated
+        in parallel, memory instructions on the left hand
+        side will be inserted into the start of the list.
+        Then, memory instruction on the LHS will be
+        transformed firstly.
+
+        :param assignblk: assignblock
+        :return: sorted list of expressions
+        """
+        instructions = []
+        for dst in assignblk:
+            # dst = src
+            aff = m2_expr.ExprAff(dst, assignblk[dst])
+            # insert memory expression into start of list
+            if isinstance(dst, m2_expr.ExprMem):
+                instructions.insert(0, aff)
+            else:
+                instructions.append(aff)
+
+        return instructions
+
+    @staticmethod
+    def _convert_block(ib, l):
+        """
+        Transforms an IRA block inplace into SSA
+        :param ib: IRA block to be transformed
+        :param l: list of SSA expressions
+        """
+        # iterator over SSA expressions
+        ssa_iter = iter(l)
+        # walk over IR blocks' assignblocks
+        for index, assignblk in enumerate(ib.irs):
+            # list of instructions
+            instructions = []
+            # insert SSA instructions
+            for dst in assignblk:
+                instructions.append(ssa_iter.next())
+            # replace instructions of assignblock in IRA block
+            ib.irs[index] = AssignBlock(instructions)
+
+    def _copy_block(self, block):
```

Comment:
Idem, `block` is quite misleading here. You can use `label` instead.

As this function is more related to `ir`, you can move it in the class `ir`.